pytorch_lightning/profiler/profilers.py

```diff
@@ -543,3 +550,262 @@ def __del__(self):
         """Close profiler's stream."""
         if self.output_file:
             self.output_file.close()
+
+
+class ScheduleWrapper:
+    """
+    This class is used to override the schedule logic from the profiler and perform
+    recording for both `training_step`, `validation_step`.
+    """
+
+    def __init__(self, schedule: Optional[Callable]):
```

Comment:
`schedule` is not supposed to be Optional, right?

Reply:
Good catch !